leaf-ui/scripts/main.js

```diff
@@ -47,6 +49,14 @@ mode = "Modelling";		// 'Analysis' or 'Modelling'
 linkMode = "Relationships";	// 'Relationships' or 'Constraints'
 
 graph = new joint.dia.Graph();
```

Comment:
This one I did when I was refactoring the code. If you look the graphObject it has the same information that we are able to get from the graph (joint.dia.Graph) and in the end we were having to make updates every time the graph object changed. Since the old code was getting information from the graphObject it was getting values not up to date, and as in javascript objects you can add attributes dinamically, we can do that direct in the graph object as it is being done in a lot of parts of the code.